### PR TITLE
First pass at aligning with `cf create-service-broker`

### PR DESCRIPTION
**Pull request by duglin**
Signed-off-by: Doug Davis <dug@us.ibm.com>

### CODE REVIEW

File: k8s/service_controller/model/service_broker.go [hunk 1]
```diff
@@ -0,0 +1,11 @@
+package model
+
+type ServiceBroker struct {
+	// json info from create docs
+	// http://apidocs.cloudfoundry.org/239/service_brokers/create_a_service_broker.html
+	Name         string `json:"name"`
+	BrokerURL    string `json:"broker_url"`
```
Comment (MHBauer) [inline k8s/service_controller/model/service_broker.go]: is there going to be any negotiation on this? It seems overly verbose to call something 'broker_url' in a 'broker' object.

Comment (duglin) [inline k8s/service_controller/model/service_broker.go]: I think Google is listing some things they would like to change, but my assumption is that for the immediate future we should align with what's in the CF docs/specs just to get things working - then we can see how people feel about changing things.

File: k8s/service_controller/Makefile [hunk 2]
```diff
@@ -24,12 +24,25 @@ test-unit: binary
...
+	  sh -c "go get -d ./... && make --no-print-directory test-unit-local"
 	
 test-unit-local:
-	@TEST_DIRS=$(shell find . -name \*_test.go -exec dirname {} \; | sort -u)
+	@TEST_UNIT_DIRS="$(shell find . -name \*_test.go -exec dirname {} \; | sort -u)"
 	@echo "\n### Running the unit tests"
-	@go test $(TEST_DIRS)
+	@go test $(TEST_UNIT_DIRS)
+
+test: binary
```
Comment (MHBauer) [inline k8s/service_controller/Makefile]: why not `test-docker` and `test`? I would expect more people to have go installed than docker.

File: k8s/service_controller/server/controller.go [hunk 3]
```diff
@@ -63,18 +63,20 @@ func (c *Controller) Inventory(w http.ResponseWriter, r *http.Request) {
...
-	var b ServiceBroker
+	var b model.ServiceBroker
 	err := utils.BodyToObject(r, &b)
 	if err != nil {
 		fmt.Printf("Error unmarshaling: %#v", err)
 		utils.WriteResponse(w, 400, err)
 		return
 	}
 
+	fmt.Printf("REQ: %v\n", b)
```
Comment (MHBauer) [inline k8s/service_controller/server/controller.go]: this leads to another issue/potential pr. We need a logging service/library.

File: k8s/service_controller/tests/sb_create_test.go [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+
+import (
+	"testing"
+
+	"bytes"
+	"net/http"
+	"os/exec"
+)
+
+func Test_SB_create_basic(t *testing.T) {
```
Comment (MHBauer) [inline k8s/service_controller/tests/sb_create_test.go]: delete or uncomment

Comment (MHBauer) [inline k8s/service_controller/tests/sb_create_test.go]: in fact, you can get rid of this whole test file if this doesn't get uncommented.

File: k8s/service_controller/Makefile (hunk 2 above)
Comment (duglin) [inline k8s/service_controller/Makefile]: let's deal with that in a separate PR - I want to get some of these re-org changes in.

**Comment by duglin** [comment]
LGTM

**Pull request merged**